cpu/riscv_common/Kconfig

```diff
@@ -12,9 +12,11 @@ config CPU_ARCH_RISCV
     select HAS_NEWLIB
     select HAS_PERIPH_CORETIMER
     select HAS_PICOLIBC if '$(RIOT_CI_BUILD)' != '1'
-    select MODULE_MALLOC_THREAD_SAFE if TEST_KCONFIG
     select HAS_SSP
 
+    select MODULE_MALLOC_THREAD_SAFE if TEST_KCONFIG
+    imply MODULE_NEWLIB_NANO
```

Comment:
Isn't `TEST_KCONFIG` missing?

Reply:
Imply will not force it, if `TEST_KCONFIG` is not there (or some other dependency for that matter), the symbol is not enabled.

Reply:
That's why this also works when picolibc is selected, newlib_nano would not be enabled in that case.